src/sage/rings/number_field/order.py

```diff
@@ -119,6 +119,83 @@ def quadratic_order_class_number(disc):
         h = pari.qfbclassno(disc)
     return ZZ(h)
 
+def quadratic_order_approximate_class_number(disc, *, bound=10**4):
+    r"""
+    Return *an approximation of* the class number of
+    the quadratic order of given discriminant.
+
+    Currently only implemented for maximal orders
+    in imaginary-quadratic fields.
+
+    EXAMPLES::
+
+        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
+        sage: QuadraticField(-419).class_number()
+        9
+        sage: quadratic_order_approximate_class_number(-419)
+        9.0...
+
+    ::
+
+        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
+        sage: d = 100000000000031
+        sage: QuadraticField(-d).class_number(proof=False)
+        14414435
+        sage: round(quadratic_order_approximate_class_number(-d))
+        144...
+        sage: round(quadratic_order_approximate_class_number(-d, bound=10**6))
+        1441...
+
+    ::
+
+        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
+        sage: # Test it against the exact class number computed for the CSIDH-512 prime
+        sage: # Source: https://eprint.iacr.org/2019/498.pdf
```

Comment:
```suggestion
    Test it against the exact class number computed for the CSIDH-512 prime (source: https://eprint.iacr.org/2019/498.pdf)::

        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
```

as long as it's valid reStructuredText, it's fine.